src/runtime/object.cpp

```diff
@@ -771,6 +776,15 @@ class task_manager {
         enqueue_core(t);
     }
 
+    void resolve(lean_task_object * t, object * v) {
+        unique_lock<mutex> lock(m_mutex);
+        if (t->m_value || !t->m_imp) {
```

Comment:
When is the second alternative reachable?

Reply:
It's not (I hope you mean the `!t->m_imp`).